SharpPcap/LibPcap/PcapInterface.cs

```diff
@@ -85,6 +85,7 @@ internal PcapInterface(pcap_if pcapIf, NetworkInterface networkInterface, ICrede
             Description = pcapIf.Description;
             Flags = pcapIf.Flags;
             Addresses = new List<PcapAddress>();
+            GatewayAddresses = new List<IPAddress>();
```

Comment:
I'm not following the below code changes in PcapInterface.cs. Is this saving an object creation or an assignment? It looks like good cleanup, and if that's it imo it should be its own commit separate from the tests.

Reply:
This change makes `GatewayAddresses` return an empty list instead of null when the interface have no gateway addresses.